x/custom/model.go

```diff
@@ -0,0 +1,103 @@
+package custom
+
+import (
+	"github.com/iov-one/tutorial/morm"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ morm.Model = (*CustomStateIndexed)(nil)
+
+func (m *CustomStateIndexed) SetID(id []byte) error {
+	m.ID = id
+	return nil
+}
+
+func (m *CustomStateIndexed) Validate() error {
+	var errs error
+
+	errs = errors.AppendField(errs, "Metadata", m.Metadata.Validate())
+	errs = errors.AppendField(errs, "ID", isGenID(m.ID, true))
+	errs = errors.AppendField(errs, "CustomString", customStringValidation(m.CustomString))
+	if m.CustomByte == nil {
+		errs = errors.Append(errs, errors.Field("CustomByte", errors.ErrEmpty, "missing custom byte"))
+	}
+	if m.InnerStateEnum != InnerStateEnum_CaseOne && m.InnerStateEnum != InnerStateEnum_CaseTwo {
+		errs = errors.Append(errs,
+			errors.Field("InnerStateEnum", errors.ErrState, "invalid inner state enum"))
+	}
+
+	if err := m.DeletedAt.Validate(); err != nil {
+		errs = errors.AppendField(errs, "DeletedAt", m.DeletedAt.Validate())
+	} else if m.DeletedAt == 0 {
+		errs = errors.Append(errs,
+			errors.Field("DeletedAt", errors.ErrEmpty, "missing created at"))
+	}
+	// TODO add custom validation for your state fields
```

Comment:
I prefer not to leave _TODO_ in code. I think this can be improved in one of two ways, depending on the purpose of it.
1. If this is a missing functionality that is needed, explain *why* it is missing. Suggested implementation might expire over time.
2. If this is a hint for the user that this is the place where more validation will come, than this is not something that is missing (a _TODO_). I think you can either leave a comment that this is a good place to add more validation or remove this comment completely as this should be obvious to the weave users.

Reply:
You're right. Better to remove it